src/comm/LinkInterface.h

```diff
@@ -59,27 +60,33 @@ class LinkInterface : public QThread
         _deletedByLinkManager(false)
     {
         // Initialize everything for the data rate calculation buffers.
-        inDataIndex = 0;
+        inDataIndex  = 0;
         outDataIndex = 0;
 
-        // Initialize our data rate buffers manually, cause C++<03 is dumb.
-        for (int i = 0; i < dataRateBufferSize; ++i)
-        {
-            inDataWriteAmounts[i] = 0;
-            inDataWriteTimes[i] = 0;
-            outDataWriteAmounts[i] = 0;
-            outDataWriteTimes[i] = 0;
-        }
+        // Initialize our data rate buffers.
+        memset(inDataWriteAmounts, 0, sizeof(inDataWriteAmounts));
+        memset(inDataWriteTimes,   0, sizeof(inDataWriteTimes));
+        memset(outDataWriteAmounts,0, sizeof(outDataWriteAmounts));
+        memset(outDataWriteTimes,  0, sizeof(outDataWriteTimes));
 
         qRegisterMetaType<LinkInterface*>("LinkInterface*");
     }
 
+    /**
+     * @brief Destructor
+     * LinkManager take ownership of Links once they are added to it. Once added to LinkManager
+     * use LinkManager::deleteLink to remove if necessary.
+     **/
     virtual ~LinkInterface() {
-        // LinkManager take ownership of Links once they are added to it. Once added to LinkManager
-        // user LinkManager::deleteLink to remove if necessary/
         Q_ASSERT(!_ownedByLinkManager || _deletedByLinkManager);
```

Comment:
Is the comment above no longer true or different now? Reason I put it there is because the assert (and the whole link ownership thing) is a bit strange and wanted to provide context if someone hit it.